### ISSUE TITLE
devkitPro buildscript cleanup

### ISSUE THREAD

**Issue by asiekierka** [AliceLR/megazeux#131]
* WinterMute points out $DEVKITA64 is not "a thing" and should not be used or defined
* Any and all references to 3ds_portlibs (portlibs.sh, commented out pkg-build lines) can now be safely removed.

**Comment by WinterMute** [comment]
also here https://github.com/AliceLR/megazeux/blob/319affdf18a014ba36a69dd9464f6d778a62d327/arch/3ds/Makefile.in#L37

you shouldn't add -L$(DEVKITARM)/arm-none-eabi/lib/armv6k/fpu manually, allow the linker to take care of it.

https://github.com/AliceLR/megazeux/blob/319affdf18a014ba36a69dd9464f6d778a62d327/arch/3ds/Makefile.in#L17-L20 are already defined through 3ds_rules. Defining them manually may break the build with future updates.

If you're automating builds then please use the docker images at https://hub.docker.com/u/devkitpro/ rather than hammering the release system with unnecessary downloads.

**Comment by asiekierka** [comment]
Oh, anything regarding arch/ is mostly on me. I'll look into this once I'm done tackling a breaking bug that popped up regarding directory traversal on 3DS.

**Comment by AliceLR** [comment]
All of our release builds are currently manual.  If by 'unnecessary downloads' you mean the `DK1` and `DK2` scripts in [this folder](https://github.com/AliceLR/megazeux/tree/master/scripts/msys2-devkitpro-scripts) (and the shell scripts they run), those particular scripts are actually intended to be run only once when the MSYS2 environment is initialized. Sorry for the confusion there!

Re: `$DEVKITA64`, is there currently a preferred alternative to that? I think anywhere I added that was only so that the Switch scripts would take roughly the same arguments as the other platform scripts.

**Comment by WinterMute** [comment]
@AliceLR 

Yes, I did mean the DK1 and DK2 scripts, sorry, at first glance it looked like it was part of automated building.

For devkitA64 we're just basing everything off $(DEVKITPRO) rather than having an extra variable. This was really just an artifact of starting with devkitARM and building out the rest of the infrastructure from there and we decided to clean things up a bit and re-organise for the switch toolset.

My main concern is that people end up cargo culting $DEVKITA64 and creating build systems based on something that doesn't exist in a standard install. Often the first we hear of it is when someone tries to build something with a fresh toolchain install and come complaining about things being broken.

With sections like this it's probably reasonable to generate $DEVKITARM rather than depending on it being passed in https://github.com/AliceLR/megazeux/blob/319affdf18a014ba36a69dd9464f6d778a62d327/scripts/msys2-devkitpro-scripts/dk-3ds-build.sh#L10-L13

so either export DEVKITARM=$DEVKITPRO/devkitARM here or just export PATH="$PATH:$DEVKITPRO/devkitARM/bin" - bear in mind that it's not recommended to swap toolchains by changing that $DEVKITARM variable since libraries compiled with one version may not be compatible with another.

**Comment by AliceLR** [comment]
I removed most reassigning of the environment variables in 2ae8923c and merged asie's fixes to the 3DS Makefile and code 08b96ba6.  I think that puts pretty much everything mentioned so far resolved.
